notebooks/bo_sphere.ipynb

```diff
@@ -108,39 +109,63 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "def logmap(x, x0):\n",
+    "    \"\"\"\n",
+    "    This functions maps a point lying on the sphere manifold into the tangent space of a second point of the manifold.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    :param x: point on the sphere manifold\n",
+    "    :param x0: basis point of the tangent space where x will be mapped\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    :return: u: vector in the tangent space of x0\n",
+    "    \"\"\"\n",
+    "    if x0.ndim < 2:\n",
+    "        x0 = x0[None]\n",
+    "\n",
+    "    if x.ndim < 2:\n",
+    "        x = x[None]\n",
+    "\n",
+    "    theta = torch.arccos(torch.maximum(torch.minimum(torch.inner(x0, x), torch.ones(1)), -torch.ones(1)))\n",
+    "    u = (x - x0 * torch.cos(theta)) * theta / torch.sin(theta)\n",
+    "\n",
+    "    u[:, torch.where(theta[0] < 1e-16)[0]] = 0.\n",
+    "\n",
+    "    return u\n",
+    "\n",
     "def ackley_function_sphere(x):\n",
     "    # Data to numpy\n",
-    "    torch_type = x.dtype\n",
-    "    x = x.cpu().detach().numpy()\n",
-    "    if np.ndim(x) < 2:\n",
+    "    if x.ndim < 2:\n",
     "        x = x[None]\n",
     "\n",
     "    # Dimension of the manifold\n",
     "    dimension = x.shape[-1]\n",
     "\n",
     "    # Projection in tangent space of the mean.\n",
     "    # The base is fixed at (1, 0, 0, ...) for simplicity. Therefore, the tangent plane is aligned with the axis x.\n",
-    "    # The first coordinate of x_proj is always 0, so that vectors in the tangent space can be expressed in a\n",
-    "    # dimension-1 dimensional space by simply ignoring the first coordinate.\n",
-    "    base = np.zeros((1, dimension))\n",
+    "    # The first coordinate of x_proj is always 0, so that vectors in the tangent space can be expressed in a dim-1\n",
```

Comment:
Maybe write "\`dimension-1\`" instead of "dim-1"? The variable we are talking about is called `dimension`, not `dim`.